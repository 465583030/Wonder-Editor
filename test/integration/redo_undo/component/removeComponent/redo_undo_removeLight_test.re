open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: remove light component", () => {
    let sandbox = getSandboxDefaultVal();
    beforeEach(() => sandbox := createSandbox());
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));
    let _simulateRemoveSpecificComponent = () =>
      SceneTreeNodeDomTool.OperateDefaultScene.getLightComponentFromDirectionLight()
      |> OperateComponentEventTool.removeComponentFromCurrentGameObject;
    let _beforeEach = () => {
      MainEditorSceneTool.initState(~sandbox, ());
      MainEditorSceneTool.createDefaultScene(
        sandbox,
        MainEditorSceneTool.setDirectionLightGameObjectTobeCurrentSceneTreeNode,
      );

      CurrentSelectSourceEditorService.setCurrentSelectSource(
        EditorType.SceneTree,
      )
      |> StateLogicService.getAndSetEditorState;
    };
    let _afterEach = () => ();

    RedoUndoTool.testRedoUndoOneStep(
      sandbox,
      "prepare first step: set currentSceneTreeNode",
      (_simulateRemoveSpecificComponent, _beforeEach, _afterEach),
      BuildComponentForRedoUndoTool.buildInspectorComponent,
    );
  });