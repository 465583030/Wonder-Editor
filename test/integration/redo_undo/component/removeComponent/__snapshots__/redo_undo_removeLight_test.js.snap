// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: remove light component prepare first step: set currentSceneTreeNode test redo operate test redo one step undo step which from first to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-sceneTree"
  >
    <div>
      <article
        className="wonder-string-input"
      >
        <span
          className="component-label"
        >
          Name : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="Direction Light"
        />
      </article>
    </div>
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
        >
          Transform
        </div>
      </div>
      <article
        className="wonder-inspector-transform"
      >
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            position : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="30"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="4"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="10"
              />
            </article>
          </article>
        </div>
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            rotate : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="45"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
          </article>
        </div>
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            scale : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
          </article>
        </div>
      </article>
    </article>
    <article
      className="wonder-addable-component"
    >
      <div
        className="addable-component-content"
      >
        <button
          className="addable-btn"
          onClick={[Function]}
        >
          Add Component
        </button>
      </div>
    </article>
  </article>
</article>
`;

exports[`redo_undo: remove light component prepare first step: set currentSceneTreeNode test undo operate test not undo 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-sceneTree"
  >
    <div>
      <article
        className="wonder-string-input"
      >
        <span
          className="component-label"
        >
          Name : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="Direction Light"
        />
      </article>
    </div>
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
        >
          Transform
        </div>
      </div>
      <article
        className="wonder-inspector-transform"
      >
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            position : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="30"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="4"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="10"
              />
            </article>
          </article>
        </div>
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            rotate : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="45"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
          </article>
        </div>
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            scale : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
          </article>
        </div>
      </article>
    </article>
    <article
      className="wonder-addable-component"
    >
      <div
        className="addable-component-content"
      >
        <button
          className="addable-btn"
          onClick={[Function]}
        >
          Add Component
        </button>
      </div>
    </article>
  </article>
</article>
`;

exports[`redo_undo: remove light component prepare first step: set currentSceneTreeNode test undo operate test undo one step undo step which from first to zero 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-sceneTree"
  >
    <div>
      <article
        className="wonder-string-input"
      >
        <span
          className="component-label"
        >
          Name : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="Direction Light"
        />
      </article>
    </div>
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
        >
          Transform
        </div>
      </div>
      <article
        className="wonder-inspector-transform"
      >
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            position : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="30"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="4"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="10"
              />
            </article>
          </article>
        </div>
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            rotate : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="45"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="0"
              />
            </article>
          </article>
        </div>
        <div
          className="transform-item"
        >
          <div
            className=""
          >
            scale : 
          </div>
          <article
            className="wonder-transform-template"
          >
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                X : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Y : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
            <article
              className="wonder-float-input"
            >
              <span
                className="component-label"
              >
                Z : 
              </span>
              <input
                className="input-component float-input"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="1"
              />
            </article>
          </article>
        </div>
      </article>
    </article>
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
        >
          Light
        </div>
        <span
          className="header-close"
          onClick={[Function]}
        >
          x
        </span>
      </div>
      <article
        className="wonder-light"
      >
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              type : 
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="0"
              >
                direction_light
              </option>
              <option
                value="1"
              >
                point_light
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-direction-light"
          >
            <article
              className="wonder-material-color"
            >
              <div
                className=""
              >
                <span
                  className=""
                >
                  color : 
                </span>
                <span
                  className=""
                >
                  #ffffff
                </span>
                <button
                  className=""
                  onClick={[Function]}
                >
                  pick color
                </button>
              </div>
            </article>
            <div
              className="light-intensity"
            >
              <article
                className="wonder-floatInput-base"
              >
                <article
                  className="wonder-float-input"
                >
                  <span
                    className="component-label"
                  >
                    Intensity : 
                  </span>
                  <input
                    className="input-component float-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                    value="1"
                  />
                </article>
              </article>
            </div>
          </article>
        </div>
      </article>
    </article>
    <article
      className="wonder-addable-component"
    >
      <div
        className="addable-component-content"
      >
        <button
          className="addable-btn"
          onClick={[Function]}
        >
          Add Component
        </button>
      </div>
    </article>
  </article>
</article>
`;
